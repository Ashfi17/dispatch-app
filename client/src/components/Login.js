import React, { Component } from "react";
import { login } from "../actions/Users";
import { connect } from "react-redux";

import "../assets/styles/login.css";

export class Login extends Component {
  state = {
    name: "",
    email: "",
    errorMessage: "",
  };
  //function to handle the text change
  OnTextChange = (event) => {
    this.setState({
      errorMessage: "",
      [event.target.name]: event.target.value,
    });
  };

  //function to login the user by passing the values typed by user in input field
  loginUser = async () => {
    let userLoginCreds = {
      user_email: this.state.email,
      user_password: this.state.password,
    };
    await this.props.login(userLoginCreds, this.props.history);
    this.props.errorMessage
      ? this.setState({ errorMessage: this.props.errorMessage })
      : this.setState({ email: "", password: "" });
  };

  render() {
    return (
      <div>
        <p
          style={{
            textAlign: "center",
          }}
        >
          {this.props.history.location.state
            ? this.props.history.location.state.message
            : ""}
        </p>
        <div className="login-signup-card">
          <div className="login-signup-elements-container">
            <h3 className="login-Signup-Text-header">Login</h3>
            <input
              id="login-email"
              className="input-field"
              name="email"
              placeholder="enter email"
              value={this.state.email}
              onChange={(e) => this.OnTextChange(e)}
            />
            <input
              id="login-password"
              className="input-field"
              name="password"
              placeholder="enter password"
              type="password"
              value={this.state.password}
              onChange={(e) => this.OnTextChange(e)}
            />
            <span className="error-text">{this.state.errorMessage}</span>
            <button
              id="login-submit-button"
              className="submit-button"
              onClick={this.loginUser}
            >
              Login
            </button>
            <span
              style={{
                textAlign: "center",
                cursor: "pointer",
                marginTop: "8px",
              }}
              onClick={() => this.props.history.push("/sign-up")}
            >
              Dont have an account ? SignUp
            </span>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  loginMessage: state.UserReducer.loginMessage,
  errorMessage: state.UserReducer.errorMessage,
});

export default connect(mapStateToProps, { login })(Login);
