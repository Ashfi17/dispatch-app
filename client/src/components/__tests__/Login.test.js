import React from "react";
import { shallow } from "enzyme";
import { Login } from "../Login";

const login = jest.fn();
const history = {
  push: jest.fn(),
  location: { state: { message: "Token Expired please login again" } },
};

const wrapper = shallow(<Login login={login} history={history} />);

describe("should test the Login component", () => {
  it("should test if the component renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should check if the text is being typed and the onTextChange function is called", () => {
    const OnTextChange = jest.spyOn(wrapper.instance(), "OnTextChange");

    const email = {
      target: "dummy email",
    };

    const password = {
      target: "dummy password",
    };

    wrapper.find("#login-email").simulate("change", email);
    expect(OnTextChange).toBeCalled();

    wrapper.find("#login-password").simulate("change", password);
    expect(OnTextChange).toBeCalled();
  });
  it("should check if the action is called if all the condtions are validated", () => {
    const loginUser = jest.spyOn(wrapper.instance(), "loginUser");
    wrapper.setState({
      name: "test name",
      email: "dummy@gmail.com",
      address: "address line 123",
      password: "abc1234",
      confirmPassword: "abc1234",
    });
    wrapper.find("#login-submit-button").simulate("click");
    expect(loginUser).toBeCalled();
    expect(wrapper.state().errorMessage).toBe("");
    expect(login).toBeCalledWith(
      {
        user_email: "dummy@gmail.com",

        user_password: "abc1234",
      },
      history
    );
  });
});
