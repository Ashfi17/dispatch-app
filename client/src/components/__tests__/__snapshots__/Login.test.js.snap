// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Login component should test if the component renders properly 1`] = `
<div>
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Token Expired please login again
  </p>
  <div
    className="login-signup-card"
  >
    <div
      className="login-signup-elements-container"
    >
      <h3
        className="login-Signup-Text-header"
      >
        Login
      </h3>
      <input
        className="input-field"
        id="login-email"
        name="email"
        onChange={[Function]}
        placeholder="enter email"
        value=""
      />
      <input
        className="input-field"
        id="login-password"
        name="password"
        onChange={[Function]}
        placeholder="enter password"
        type="password"
      />
      <span
        className="error-text"
      />
      <button
        className="submit-button"
        id="login-submit-button"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
